python/bench/upload_examples_bench.py

```diff
@@ -0,0 +1,124 @@
+import statistics
+import time
+from typing import Dict
+from uuid import uuid4
+from langsmith.schemas import DataType, ExampleCreateWithAttachments
+import sys
+sys.path.append('./../langsmith')
+from client import Client
+
+def create_large_json(length: int) -> Dict:
+    """Create a large JSON object for benchmarking purposes."""
+    large_array = [
+        {
+            "index": i,
+            "data": f"This is element number {i}",
+            "nested": {"id": i, "value": f"Nested value for element {i}"},
+        }
+        for i in range(length)
+    ]
+
+    return {
+        "name": "Huge JSON" + str(uuid4()),
+        "description": "This is a very large JSON object for benchmarking purposes.",
+        "array": large_array,
+        "metadata": {
+            "created_at": "2024-10-22T19:00:00Z",
+            "author": "Python Program",
+            "version": 1.0,
+        },
+    }
+
+
+def create_example_data(dataset_id: str, json_size: int) -> Dict:
+    """Create a single example data object."""
+    return ExampleCreateWithAttachments(**{
+        "dataset_id": dataset_id,
+        "inputs": create_large_json(json_size),
+        "outputs": create_large_json(json_size),
+    })
+
+DATASET_NAME = "TEST DATASET"
+def benchmark_example_uploading(num_examples: int, json_size: int, samples: int = 1) -> Dict:
+    """
+    Benchmark run creation with specified parameters.
+    Returns timing statistics.
+    """
+    multipart_timings, old_timings = [], []
+
+
+    for _ in range(samples):
+        client = Client(api_url="https://dev.api.smith.langchain.com")
+
+        if client.has_dataset(dataset_name=DATASET_NAME):
+            client.delete_dataset(dataset_name=DATASET_NAME)
+
+        dataset = client.create_dataset(
+            DATASET_NAME,
+            description="Test dataset for multipart example upload",
+            data_type=DataType.kv,
+        )
+        examples = [create_example_data(dataset.id, json_size) for i in range(num_examples)]
+
+        # Old method
+        old_start = time.perf_counter()
+        inputs=[e.inputs for e in examples]
+        outputs=[e.outputs for e in examples]
+        # the create_examples endpoint fails above 20mb
+        try:
+            client.create_examples(inputs=inputs,
+                                outputs=outputs,dataset_id=dataset.id)
+            old_elapsed = time.perf_counter() - old_start
+        except:
+            old_elapsed = 1000000
```

Comment:
would just remove the try except. If the code throws, then the benchmark should just exit